compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp

```diff
@@ -2529,12 +2533,28 @@ LogicalResult initGPULaunchConfig(FunctionOpInterface funcOp) {
         break;
       }
     }
+
+    if (auto scatterOp = dyn_cast<IREE::LinalgExt::ScatterOp>(op)) {
+      auto indices = scatterOp.getIndices();
+      if (indices.getDefiningOp() == nullptr) {
+        continue;
+      }
+      // If scatter's indices is a generic op, mark it as to skip.
+      if (auto genericOp =
+              dyn_cast<linalg::GenericOp>(indices.getDefiningOp())) {
+        genericToSkip.insert(genericOp);
+      }
+      // Mark scatter's backward slices as to skip too.
+      SetVector<Operation *> slices;
+      getBackwardSlice(indices, &slices);
+      genericToSkip.insert(slices.begin(), slices.end());
```

Comment:
nit: you can use `append_range`
```suggestion
      llvm::append_range(genericToSkip, slices);
```

Reply:
I just gave it a shot, as it turns out `llvm::append_range` is not compatible with `llvm::SmallDenseSet`.

```cpp
template <typename Container, typename Range>
void append_range(Container &C, Range &&R) {
  C.insert(C.end(), adl_begin(R), adl_end(R));
}
```

`append_range` expects the container to support `insert` with 3 iterators. Clang complained that it can't find matching function and can only find insert with 1 or 2 arguments:
```
candidate function template not viable: requires 2 arguments, but 3 were provided
    template <typename InputIt> void insert(InputIt I, InputIt E) {
/root/iree/third_party/llvm-project/llvm/include/llvm/ADT/DenseSet.h:213:29: note: candidate function not viable: requires single argument 'V', but 3 arguments were provided
    std::pair<iterator, bool> insert(const ValueT &V) {
/root/iree/third_party/llvm-project/llvm/include/llvm/ADT/DenseSet.h:218:29: note: candidate function not viable: requires single argument 'V', but 3 arguments were provided
   std::pair<iterator, bool> insert(ValueT &&V) {
```

Nice intro to some handy utilities though!